src/api/fs.go

```diff
@@ -0,0 +1,131 @@
+package api
+
+import (
+	"os"
+	"regexp"
+	"strings"
+)
+
+func PathExists(path string) bool {
+	_, err := os.Stat(path)
+	return !os.IsNotExist(err)
+}
+
+func FileExists(path string, isDirectory bool) bool {
+	info, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		return false
+	}
+	if isDirectory && info.IsDir() || !isDirectory && !info.IsDir() {
+		return true
+	}
+
+	return false
+}
+
+func ReadFile(path string) (string, error) {
+	file, err := os.ReadFile(path)
+	if err != nil {
+		return "", err
+	}
+	return string(file), nil
+}
+
+func ReadDir(path string) ([]os.DirEntry, error) {
+	return os.ReadDir(path)
+}
+
+func removeRanges(file string, rangeStart string, rangeEnd string, removeRangeEnd bool) string {
+	cleanFile := ""
+	slice := file
+	sliceLen := len(slice)
+	endLen := 0
+	if removeRangeEnd {
+		endLen = len(rangeEnd)
+	}
+
+	for {
```

Comment:
I propose changing the loop to something like this:
```go
commentPos := strings.Index(slice, rangeStart)
for commentPos != -1 {
    cleanFile += slice[0:commentPos]
    slice = slice[commentPos:sliceLen]
    sliceLen = len(slice)
    realEndPos := strings.Index(slice, rangeEnd) + endLen
    if realEndPos == -1 {
        realEndPos = len(slice)
    }

    slice = slice[realEndPos:sliceLen]
    sliceLen = len(slice)
    commentPos = strings.Index(slice, rangeStart)
}
```
This way the first if statement inside the loop can be omitted. The loop breaks at the specified condition anyway so let's use this "while loop"-like structure